src/PointwiseFunctions/Elasticity/ConstitutiveRelations/Tags.hpp

```diff
@@ -3,37 +3,52 @@
 
 #pragma once
 
-#include <string>
-
 #include "DataStructures/DataBox/Tag.hpp"
-#include "DataStructures/Tensor/TypeAliases.hpp"
-#include "PointwiseFunctions/Elasticity/ConstitutiveRelations/ConstitutiveRelation.hpp"
-
-/// \cond
-namespace Elasticity {
-namespace Tags {
-template <size_t Dim>
-struct Stress;
-}  // namespace Tags
-}  // namespace Elasticity
-namespace Tags {
-template <size_t Dim, typename Frame>
-struct Coordinates;
-}  // namespace Tags
-/// \endcond
+#include "Utilities/TMPL.hpp"
 
 namespace Elasticity {
 namespace Tags {
 
 /// Base tag for the constitutive relation
 struct ConstitutiveRelationBase : db::BaseTag {};
 
+/*!
+ * \brief The elastic material's constitutive relation.
+ *
+ * \see `Elasticity::ConstitutiveRelations::ConstitutiveRelation`
+ */
 template <typename ConstitutiveRelationType>
 struct ConstitutiveRelation : ConstitutiveRelationBase, db::SimpleTag {
-  static constexpr OptionString help = {
-      "The constitutive relation of the elastic material"};
   using type = ConstitutiveRelationType;
-  static std::string name() noexcept { return "Material"; }
+};
+
+/*!
+ * \brief The elastic material's constitutive relation, loaded from options in
+ * the `ProviderOptionTag`.
+ *
+ * Retrieves the constitutive relation from the object constructed from the
+ * `ProviderOptionTag` by calling its `constitutive_relation()` member function.
+ * Also requires the `ProviderOptionTag::type` to provide a
+ * `constitutive_relation_type` type alias.
+ *
+ * The constitutive relation can be retrieved from the DataBox using its base
+ * `Elasticity::Tags::ConstitutiveRelation` tag.
+ */
+template <typename ProviderOptionTag>
+struct ConstitutiveRelationFrom
+    : ConstitutiveRelation<typename tmpl::type_from<
```

Comment:
you're still having a simple tag inherit from a simple tag

Reply:
do you have another suggestion? There will be different ways of getting the `Tags::ConstitutiveRelation` into the DataBox:
- Put this `ConstitutiveRelationFrom` tag into the ConstGlobalCache so the constitutive relation is retrieved from the analytic solution or analytic data
- Load the constitutive relation directly from the input file

Reply:
Why not specify the constitutive relation in the metavariables like we do for equations of state?